pkg/registry/client.go

```diff
@@ -195,12 +191,9 @@ func ClientOptPlainHTTP() ClientOption {
 	}
 }
 
-// ClientOptResolver returns a function that sets the resolver setting on a client options set
-func ClientOptResolver(resolver remotes.Resolver) ClientOption {
-	return func(client *Client) {
-		client.resolver = func(_ registry.Reference) (remotes.Resolver, error) {
-			return resolver, nil
-		}
+func ClientOptResolver(_ remotes.Resolver) ClientOption {
+	return func(c *Client) {
+		c.err = errDeprecatedRemote
```

Comment:
I honestly think, converting the previous build time breakage to a runtime error (in the name of maintaining API compatibility) is a much worse idea.

That said, aside from deferring this PR to Helm v4, I'm not sure how to solve this others. The choices are API or functional breakage.

Reply:
More discussion on this from Zoran's PR https://github.com/helm/helm/pull/12310/files#r1499833508 I don't love it either, but I'm not seeing a better solution

Reply:
Bad for a few, but great for many.

Reply:
For Helm 4, we should just remove `ClientOptResolver` I think

For a Helm 3 backport, what would be the alternative for users? Can we document in a function doc comment?